File: 2020/1.py
import time
from utils import readInput
from itertools import combinations
from math import prod

def loadInput():
    lines = readInput("input_1.txt")
    expenses = [int(e) for e in lines]
    return expenses

def part1(expenses):
    for c in combinations(expenses, 2):
        if sum(c) == 2020:
            print ("🎄 Part 1: {}".format(prod(c)))
            break

def part2(expenses):
    for c in combinations(expenses, 3):
        if sum(c) == 2020:
            print ("🎄🎅 Part 2: {}".format(prod(c)))
            break


print("⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄")
print("⛄        Day 1         ⛄")
print("⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄")

inputs = loadInput()

t0 = time.time()
part1(inputs)
print ("Time: {:.5f}".format(time.time()-t0))

t0 = time.time()
part2(inputs)
print ("Time: {:.5f}".format(time.time()-t0))

File: 2020/1.jl
include("utils.jl")
using .AoCUtils: readInput
using Combinatorics

function loadInput()
    filename = "input_1.txt"
    lines = readInput(filename)
    expenses = parse.(Int, lines)

    return expenses
end

function part1(expenses::Vector{Int})
    for c in combinations(expenses, 2)
        if sum(c) == 2020
            println("🎄 Part 1: $(prod(c))")
            break
        end
    end
end

function part2(expenses::Vector{Int})
    for c in combinations(expenses, 3)
        if sum(c) == 2020
            println("🎄🎅 Part 2: $(prod(c))")
            break
        end
    end
end

println("⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄")
println("⛄        Day 1         ⛄")
println("⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄⛄")

expenses = loadInput()
@time part1(expenses)
@time part2(expenses)
